### PR TITLE
All `sector_classification` datasets have no duplicate values of `code` column

### LINKED ISSUE #229: nace_classification has duplicate values
Hi,
it's rather a minor issue, but nace_classification dataset has duplicate values in the code field. As a result same code may be mapped to different sectors.  

```r
library(dplyr)

#code duplicates
r2dii.data::nace_classification %>% dplyr::count(code) %>% filter(n>1)
```

### PR DESCRIPTION
**Pull request by jdhoffa**
Closes #229

### CODE REVIEW

**Comment by jdhoffa** [comment]
Looking into this now, and there are some issues:
For the `cnb_classification` file, there is a code that is mapped to a whole bunch of different sectors. My understanding is that this should not be allowed in the Banks code, as it is an indication that the "code level" is not granular enough to identify the correct sector. 

Screenshot 2024-02-05 at 15 36 52

**Comment by jdhoffa** [comment]
A similar issue presents itself in the `gics_classification` file:
Screenshot 2024-02-05 at 15 43 57

**Comment by jdhoffa** [comment]
`nace_classification` is the most complicated here. 

What happens is that `original_code` gets mapped to `code`, which removes the decimal mark. This is a bad idea as it effectively converts different codes (e.g. `6.2` and `62`) to the same "formatted" code `620`. 

This is almost certainly a bug, and should be reverted, however, we do risk affecting the downstream user as it will change the expected sector classification code. I leave it to @jacobvjk and George to determine how risky this change will end up being:


Screenshot 2024-02-05 at 15 48 00

**Comment by jdhoffa** [comment]
`cnb_classification` - @george votes to drop this dataset entirely 
`gics_classification` - @george and @jacobvjk both agree to remove "coal" entirely from the 151040 code
`isic_classification` - @george and @jacobvjk think this dataset is messed up, so they want to drop it. Separately, we should have an automated warning message in `r2dii.match` that suggests users to e-mail us/ or go to a link to GitHub issues where they can request what they want.
`nace_classification` - Use the `original_code` AND use the letter prepended. (e.g. "A1.11"). They think that `original_code` should map to `code`, and that we should drop the `code_level` column entirely.

**Comment by jdhoffa** [comment]
This is ready for review. 

Progress on the deprecation of `cnb_classification` and `isic_classification` will be tracked in #329

File: data-raw/classification_bridge.R [hunk 1]
```diff
@@ -2,15 +2,51 @@ library(usethis)
 
 source(file.path("data-raw", "utils.R"))
 
+letter_to_number <- Vectorize(letter_to_number)
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:5]: this does not seem to be used anywhere

**Review by jacobvjk — COMMENTED** [review]
There seems to be some stray code that I could not see used anywhere else in the repo. See comment. Maybe remove?

File: data-raw/classification_bridge.R (hunk 1 above)
Comment (jdhoffa) [reply data-raw/classification_bridge.R]: Nice catch!

**Review by jacobvjk — APPROVED** [review]
lgtm

**Pull request merged**